python/cudf/cudf/core/column/decimal.py

```diff
@@ -402,15 +416,12 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
                 # to try the next dtype
                 continue
 
-    # Instead of raising an overflow error, we create a `Decimal128Dtype`
-    # with max possible scale & precision, see example of this demonstration
-    # here: https://learn.microsoft.com/en-us/sql/t-sql/data-types/
-    # precision-scale-and-length-transact-sql?view=sql-server-ver16#examples
-    scale = min(
-        scale, cudf.Decimal128Dtype.MAX_PRECISION - (precision - scale)
+    # if we've reached this point, we cannot create a decimal type without
+    # overflow; raise an informative error
+    raise ValueError(
+        f"Performing {op} between columns of type {repr(lhs_dtype)} and "
```

Comment:
Do we actually need to call `repr` here? Shouldn't the printing action trigger `__repr__` automatically?

Reply:
Think the printing triggers `__str__`, which ends up giving an error message that doesn't include the scale/precision:

```
ValueError: Performing __mul__ between columns of type decimal128 and decimal128 would result in overflow
```

Thought that forcing use of `__repr__` here to show the scale and precision would be more informative, though open to suggestions for a better error message in this case (cc @shwina if you have thoughts)

Reply:
Honestly, we should probably include the scale/precision in both str and repr. But I think changing `__str__` is too large an undertaking for this PR so I'm good with this change.